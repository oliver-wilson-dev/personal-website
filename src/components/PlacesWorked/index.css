.section {
    &Wrapper {
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        
        margin-top: 30px;
        background: var(--projectGreen);
    }

    &Container {
        padding: 30px 20px;
    }
}

.statement {
    color: var(--primaryTextColorLight);
    text-align: center;
    margin: 0 0 40px 0;
    font-size: 20px;
    font-family: 'Source Sans Pro', sans-serif;
}

.companyLogo {
    max-width: 140px;
    fill: var(--primaryTextColorLight);
}

.itvLogo {
    @extend .companyLogo;
    max-width: 130px;
}

.companyLogoWrapper {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    align-items: center;
}

@media (min-width: 0px) and (max-width: 425px) {
    .companyLogoWrapper {
        flex-direction: column;
    }

    .statement {
        font-size: 14px;
        margin: 0 0 30px 0;
    }

    .companyLogo {
        max-width: 100px;
        
        &:last-child {
            margin-top: 30px;
        }
    }

    .itvLogo {
        max-width: 100px;
    }
  }