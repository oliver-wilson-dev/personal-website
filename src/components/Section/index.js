import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import sharedStyles from '../App/index.css';
import styles from './index.css';
import useFadeInClasses from '../../hooks/useFadeInClasses';

const Section = ({ title, children, additionalStyles = {} }) => {
  const { fadeInClasses } = useFadeInClasses();

  return (
    <section className={cn(
      styles.flexColumn,
      sharedStyles.flexCenter,
      fadeInClasses,
      { [additionalStyles.section]: !!additionalStyles.section, }
    )}
    >
      {title && <h2 className={styles.sectionTitle}>{title}</h2>}
      <div className={additionalStyles.container}>
        {children}
      </div>
    </section>
  );
};

Section.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired,
  additionalStyles: PropTypes.oneOfType([
    PropTypes.shape()
  ])
};

export default Section;
