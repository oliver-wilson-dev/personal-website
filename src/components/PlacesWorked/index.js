/* eslint-disable max-len */
import React from 'react';
import styles from './index.css';
import AsosLogo from './asos-logo.svg';
import ItvLogo from './itv-logo.svg';
import Section from '../Section';

const PlacesWorked = () => (
  <Section additionalStyles={{
    section: styles.sectionWrapper,
    container: styles.sectionContainer
  }}
  >
    <h3 className={styles.statement}>
      Proudly contributing towards projects in the fashion and television industries for over four years.
    </h3>
    <div className={styles.companyLogoWrapper}>
      <AsosLogo className={styles.companyLogo} />
      <ItvLogo className={styles.itvLogo} />
    </div>
  </Section>
);

export default PlacesWorked;
