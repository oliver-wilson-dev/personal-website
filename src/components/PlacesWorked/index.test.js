import React from 'react';
import { shallow } from 'enzyme';
import PlacesWorked from '.';
import Section from '../Section';
import styles from './index.css';

jest.mock('../Section', () => {
  const Section = () => null;

  return Section;
});

jest.mock('./asos-logo.svg', () => {
  const AsosLogo = () => null;

  return AsosLogo;
});

jest.mock('./itv-logo.svg', () => {
  const ItvLogo = () => null;

  return ItvLogo;
});

jest.mock('./sainsburys-logo.svg', () => {
  const SainsburysLogo = () => null;

  return SainsburysLogo;
});

describe('PlacesWorked', () => {
  const render = () => shallow(<PlacesWorked />);

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  it('should pass the additionalStyles to the section component', () => {
    const wrapper = render();

    expect(wrapper.find(Section).prop('additionalStyles')).toEqual({
      section: styles.sectionWrapper,
      container: styles.sectionContainer
    });
  });
});
