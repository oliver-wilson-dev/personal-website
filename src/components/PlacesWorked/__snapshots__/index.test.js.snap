// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesWorked should render correctly 1`] = `
<Section
  additionalStyles={
    Object {
      "container": "sectionContainer",
      "section": "sectionWrapper",
    }
  }
>
  <h3
    className="statement"
  >
    Proudly contributing towards projects in the fashion and television industries for over four years.
  </h3>
  <div
    className="companyLogoWrapper"
  >
    <AsosLogo
      className="companyLogo"
    />
    <ItvLogo
      className="itvLogo"
    />
  </div>
</Section>
`;
